Themis/Views/Assessment/CodeEditor/CodeView.swift

```diff
@@ -12,14 +13,36 @@ struct CodeView: UIViewControllerRepresentable {
     func makeUIViewController(context: Context) -> ViewController {
         let viewController = ViewController()
         viewController.textView.editorDelegate = context.coordinator
-        viewController.sourceCode = vm.selectedFile?.code ?? ""
+        if let selectedFile = vm.selectedFile, let code = selectedFile.code {
+            switch selectedFile.fileExtension {
+            case .swift:
+                viewController.textView.setLanguageMode(TreeSitterLanguageMode(language: .swift), completion: { _ in
+                    viewController.textView.text = code })
+            case .java:
+                viewController.textView.setLanguageMode(TreeSitterLanguageMode(language: .java), completion: { _ in
+                    viewController.textView.text = code })
+            case .other:
+                viewController.textView.setLanguageMode(PlainTextLanguageMode(), completion: { _ in
+                    viewController.textView.text = code })
+            }
```

Comment:
Maybe this could be in a method of the ViewController so that we don't have duplicated code

Reply:
Moved it in the ViewModel 👍